lexer/lexer.go

```diff
@@ -32,6 +33,32 @@ type Position struct {
 	Column int
 }
 
+func (pos Position) Add(lines int, cols int) Position {
+	return Position{
+		Line:   pos.Line + lines,
+		Column: pos.Column + cols,
+	}
+}
+
+type PositionError struct {
```

Comment:
Hmm, I can see why you've added this (for use in both parser and resolver), but I don't love how this is a new exported type in the lexer. It's only used internally. I think we should add this in package `ast` as `ast.PositionError` instead.

Additionally, I don't love the `Errorf` method being on the position type; I'd prefer it were just a method or function in the parser and/or resolver (it's a tiny function, so it's fine if it's duplicated in both if needed). Or you could add `ast.PosErrorf(pos, format, args...)` if you want to share it.

Reply:
Yeah, I've added `ast.PosErrorf(pos, format, args...)`